server/src/middleware/User.js

```diff
@@ -21,3 +21,33 @@ export const checkUserId = async (req, res, next) => {
 
   next();
 };
+
+export const updateResumeText = async (req, res, next) => {
+  try {
+    const userId = req.get("userId");
+
+    // Start parsing the PDF in the background
+    parsePdfFromBuffer(req.file.buffer)
```

Comment:
This line doesn't wait for the database call to complete. Is it safe to call `next()` without awaiting?